server/src/main/java/org/candlepin/jackson/ConsumerTypeDeserializer.java

```diff
@@ -16,77 +16,69 @@
 
 import org.candlepin.common.exceptions.CandlepinJsonProcessingException;
 import org.candlepin.dto.api.v1.ConsumerTypeDTO;
-import org.candlepin.model.ConsumerType.ConsumerTypeEnum;
 
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
 import com.fasterxml.jackson.databind.DeserializationContext;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.JsonDeserializer;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 
-
-
 /**
- * The ConsumerTypeDeserializer handles deserialization of consumer types defined as strings or
- * objects on existing objects.
+ * Handles the deserialization of the "label" field by wrapping it in a {@link ConsumerTypeDTO} object,
+ * by handling both of the following formats: <pre> {@code "type":"value" } </pre> and
+ * <pre> {@code "type":{"label ":"value", "manifest":"value"} } </pre>.
  */
-public class ConsumerTypeDeserializer extends StdDeserializer<ConsumerTypeDTO> {
-    private static Logger log = LoggerFactory.getLogger(ConsumerTypeDeserializer.class);
+public class ConsumerTypeDeserializer extends JsonDeserializer<ConsumerTypeDTO> {
 
-    public ConsumerTypeDeserializer() {
-        this(null);
-    }
+    private static Logger log = LoggerFactory.getLogger(ConsumerTypeDeserializer.class);
 
-    public ConsumerTypeDeserializer(Class<?> valueClass) {
-        super(valueClass);
-    }
+    private static String fieldName = "label";
 
     @Override
     public ConsumerTypeDTO deserialize(JsonParser parser, DeserializationContext context)
-        throws IOException, JsonProcessingException {
+        throws IOException {
 
-        // We're expecting an object or a string
-        switch (parser.currentToken()) {
-            case START_OBJECT:
-                return this.parseTypeFromObject(parser);
+        TreeNode node = parser.readValueAsTree();
 
-            case VALUE_STRING:
-                return this.parseTypeFromString(parser);
+        if (node.isValueNode()) {
+            log.debug("Processing {} as a value node.", fieldName);
 
-            default:
-                throw new CandlepinJsonProcessingException("Unexpected consumer type format: " +
-                    parser.readValueAsTree(), parser.getCurrentLocation());
+            return parseValueNode(node);
+        }
+        else if (node.isObject()) {
+            log.debug("Processing {} as a containing object node.", fieldName);
+
+            TreeNode valueNode = node.path(fieldName);
+            if (valueNode.isMissingNode()) {
+                throw new CandlepinJsonProcessingException(
+                    "The field " + fieldName + " is missing from: " + node.asToken(),
```

Comment:
Why are we substituting the `fieldName` variable in all of these strings? It's a static variable with the value 'label' that never gets reassigned as far as I can see, so there's not point. Specifically about the exception messages, we should try to keep them the same (or similar) to before, since 'Unexpected consumer type format:' is much clearer than 'The field label is missing from: '

Reply:
Agreed, Updated the exception messages.